docker/docker_image.go

```diff
@@ -103,3 +107,50 @@ func GetRepositoryTags(ctx context.Context, sys *types.SystemContext, ref types.
 	}
 	return tags, nil
 }
+
+// GetDigest returns the image's digest. (This is a
+// backward-compatible shim method which calls the module-level
+// GetDigest)
+func (i *Image) GetDigest(ctx context.Context) (*digest.Digest, error) {
+	return GetDigest(ctx, i.src.c.sys, i.src.logicalRef)
+}
+
+// GetDigest returns the image's digest
+func GetDigest(ctx context.Context, sys *types.SystemContext, ref types.ImageReference) (*digest.Digest, error) {
+	dr, ok := ref.(dockerReference)
+	if !ok {
+		return nil, errors.Errorf("ref must be a dockerReference")
+	}
+
+	tagOrDigest, err := dr.tagOrDigest()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := newDockerClientFromRef(sys, dr, false, "pull")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create client")
+	}
+
+	path := fmt.Sprintf(manifestPath, reference.Path(dr.ref), tagOrDigest)
+	headers := map[string][]string{
+		"Accept": manifest.DefaultRequestedManifestMIMETypes,
+	}
+
+	res, err := client.makeRequest(ctx, "HEAD", path, headers, nil, v2Auth, nil)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Content-Type from manifest HEAD is %q", res.Header.Get("Content-Type"))
```

Comment:
This exists in `ensureManifestIsLoaded` to debug conversion issues and misconfigured servers; it might not be necessary here. OTOH it doesn’t hurt anything.

Reply:
I prefer to keep this for debugging purposes if it's ok with you.